content/How to change the rest-auth reset password email to a custom html template - Jorge Michelena.md

```diff
@@ -0,0 +1,151 @@
+
+This feature is provided by the rest-auth library but the email that is sent to the user is a plain text one and the link that comes in the email is that of a built-in rest-auth endpoint, which may be exactly what you want in some cases, but not in others.
+If you want to change the content of the email the process is well documented but only if you want to change it to a plain text file, in case you want to change to a custom html template then there is not much information covering how to do it, that is why we will explain it in this article.
+
+### Set up
+
+I'm going to asume that you have python, django, django rest framework and rest-auth installed as well as a project ready on which you are going to make these changes.
+In my case I made a very simple project called `project` with just one app called `api`, and the following layout:
+
+
+```
+├── api
+│   ├── __init__.py
+│   ├── apps.py
+│   ├── migrations
+│   │   ├── __init__.py
+│   └── serializers.py
+├── db.sqlite3
+├── manage.py
+├── project
+│   ├── __init__.py
+│   ├── asgi.py
+│   ├── settings.py
+│   ├── urls.py
+│   ├── widgets.py
+│   └── wsgi.py
+└── templates
+    └── registration
+        └── custom_reset_confirm.html
+```
+
+As you can see I have a `templates` folder in which I made a new folder called `registration` with a file called `custom_reset_confirm.html` inside of it.
+The context in the html email template looks like this:
+
+```html
+<html>
+  <head>
+    <meta name="viewport" content="width=device-width" />
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+    <title>Confirm reset password email</title>
+  </head>
+  <body style="text-align: center;">
+    
+    <h2>Hello!</h2>
+    <br>
+    <p><b>You're receiving this email because you requested a password reset for your user account at {{ site_name }}.</b></p>
+    <br>
+    <p> Please go to the following page and choose a new password: </p>
+    <p> {{ protocol }}://{{ domain }}{% url 'password_reset_confirm' uidb64=uid token=token %} </p>
+    <p> Your username, in case you’ve forgotten: {{ user.get_username }} </p>
+    <br>
+    <p><i> Thanks for using our site! The {{ site_name }} team.</i></p>
+  </body>
+
+</html>
+
+
+```
+
+As you may have noticed, the content is pretty much the same as in the default email, and for the sake of simplicity I didn't add a lot of style, just some bold and italic text as well as greeting in a bigger font size.
+The relevant variables like `site_name`, `domain` and such will be automatically assigned values passed on the context to the template.
+
+### Changing the default email
+
+Now that we have our template what we need to do is indicate that this is what we want to send our users when they request to reset
+their password, to do that we have to define a serializer that inherits from `rest-auth`'s `PasswordResetSerializer` and overwrite
+its `get_email_options` method and then use that serializer instead of the default one.
```

Comment:
```suggestion
Now that we have our template what we need to do is indicate that this is what we want to send to our users when they request a password reset. To do that we have to define a serializer that inherits from `dj-rest-auth`'s `PasswordResetSerializer` and overwrite
its `get_email_options` method and then use that serializer instead of the default one.
```